src/internal/components/abstract-switch/__tests__/abstract-switch.test.tsx

```diff
@@ -92,5 +93,45 @@ describe('Abstract switch', () => {
       expect(nativeControl).toHaveAccessibleName('Default label Custom label');
       expect(nativeControl).toHaveAccessibleDescription('Custom description Default description');
     });
+
+    test('does not trigger onClick when disabled', () => {
+      const onClick = jest.fn();
+      const wrapper = renderAbstractSwitch({
+        controlClassName: '',
+        outlineClassName: '',
+        styledControl: <div />,
+        controlId: 'custom-id',
+        ariaLabelledby: 'some-custom-label',
+        ariaDescribedby: 'some-custom-description',
+        label: 'Default label',
+        description: 'Default description',
+        disabled: true,
+        nativeControl: nativeControlProps => <input {...nativeControlProps} className="switch-element" type="radio" />,
+        onClick,
+      });
+
+      wrapper.findByClassName(styles['label-wrapper'])!.click();
```

Comment:
Can we not use the AbstractSwitch test util wrapper here? 

https://github.com/cloudscape-design/components/blob/c54515793103c374eb72cf76a52a618e262c960a/src/test-utils/dom/internal/abstract-switch.ts#L6

Or similar to above, wrap the render in a label and click that?

Reply:
good catch. yes, we can. updated